art/preprocessing/standardisation_mean_std/standardisation_mean_std_tensorflow.py

```diff
@@ -49,19 +54,21 @@ def __init__(
         :param std: Standard Deviation.
         """
         super().__init__(is_fitted=True, apply_fit=apply_fit, apply_predict=apply_predict)
-        self.mean = mean
-        self.std = std
+        self.mean = np.asarray(mean, dtype=ART_NUMPY_DTYPE)
+        self.std = np.asarray(std, dtype=ART_NUMPY_DTYPE)
         self._check_params()
 
     def forward(self, x: "tf.Tensor", y: Optional["tf.Tensor"] = None) -> Tuple["tf.Tensor", Optional["tf.Tensor"]]:
         """
         Apply standardisation with mean and standard deviation to input `x`.
+
+        :param x: Input samples to standardise of shapes `NCHW`, `NHWC`, `NCFHW` or `NFHWC`.
+        :param y: Label data, will not be affected by this preprocessing.
+        :return: Standardise input samples and unmodified labels.
         """
         import tensorflow as tf  # lgtm [py/repeated-import]
 
-        mean = np.asarray(self.mean, dtype=ART_NUMPY_DTYPE)
-        std = np.asarray(self.std, dtype=ART_NUMPY_DTYPE)
-
+        mean, std = broadcastable_mean_std(x, self.mean, self.std)
```

Comment:
Would it be possible to run `broadcastable_mean_std` outside of method `forward`? It should only be required once.

Reply:
I don't see how we could move the call outside `forward`, e.g. into the `__init__` method. `broadcastable_mean_std` requires knowledge of input data `x` in order to infer how to reshape `mean` and `std`.

Reply:
As discussed in our chat, I have added lazy attributes for broadcastable `mean` and `std`.